### DISCUSSION TITLE
Link pull requests (PRs) to issues across repositories

### DISCUSSION THREAD

**Discussion by 0xdevalias** [community/community#4194] [Pull Requests]
## Background

Currently it's possible to link a pull request (PR) to an issue:

- **Manually:** When both are in the same repository
- **Using closing keywords:** This works across repositories, but only when the PR is made to the default branch

This is documented here:

- https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue

## Problem

For public/private workflows (quite common in open source projects), it makes a lot of sense to be able to link PRs (eg. on a public repo) to their corresponding issue on a different repo (eg. on a private repo).

Currently we can use the 'closing keywords' to work around this when targeting the default branch which works ok most of the time, but there are edge cases and limitations where it's a suboptimal workflow. eg.

- When a PR isn't targeting the default branch, we have no way of linking it to the issue, because the closing keywords are ignored
- When we use closing keywords, the corresponding issue is closed when the PR is merged (as intended), but often that issue isn't actually 'done' when the PR is merged, and so we then have to go through and manually reopen it again, which adds needless friction + noise.

## Proposed Solutions

It would be nice if, at a minimum, we were able to:

- Manually link a PR from one repo, with an issue on another repo
- Use a 'non-closing-keyword' to link a PR to an issue (both within the same repo, and across repos) that WON'T close the issue when the PR is merged
  - eg. relates to #123, link #123, or something like this
  
## Etc

- Note: I have also raised this directly with GitHub support just now:
  - 🔒 https://support.github.com/ticket/personal/0/1197376

**Comment by alperchoco** [comment]
Yeah, this sucks. It forces us to have all of our work in a single monorepo, which I'm not against but is definitely not something that should be forced on people.

For us as platform, a lot of our work spans repos across the org and not being able to link it correctly is less than ideal.

**Comment by Alex-ley-scrub** [comment]
We need this so badly - how is it possible in the UI but not via GitHub Rest/GraphQL API?!

see meta issue: https://github.com/orgs/community/discussions/186495
